x-pack/test/security_solution_ftr/services/detections/endpoint_rule_alert_generator.ts

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { BaseDataGenerator } from '@kbn/security-solution-plugin/common/endpoint/data_generators/base_data_generator';
+import endpointPrePackagedRule from '@kbn/security-solution-plugin/server/lib/detection_engine/prebuilt_rules/content/prepackaged_rules/elastic_endpoint_security.json';
+import { kibanaPackageJson } from '@kbn/utils';
+import { mergeWith } from 'lodash';
+import { EndpointMetadataGenerator } from '@kbn/security-solution-plugin/common/endpoint/data_generators/endpoint_metadata_generator';
+import { HostMetadata } from '@kbn/security-solution-plugin/common/endpoint/types';
+import { DeepPartial } from 'utility-types';
+
+const mergeAndReplaceArrays = <T, S>(destinationObj: T, srcObj: S): T => {
+  const customizer = (objValue: T[keyof T], srcValue: S[keyof S]) => {
+    if (Array.isArray(objValue)) {
+      return srcValue;
+    }
+  };
+
+  return mergeWith(destinationObj, srcObj, customizer);
+};
+
+type EndpointRuleAlert = Pick<
+  HostMetadata,
+  'Endpoint' | 'agent' | 'elastic' | 'host' | 'data_stream'
+> & {
+  [key: string]: any;
+};
+
+export class EndpointRuleAlertGenerator extends BaseDataGenerator {
+  /** Generates an Endpoint Rule Alert document */
+  generate(overrides: DeepPartial<EndpointRuleAlert> = {}): EndpointRuleAlert {
+    const endpointMetadataGenerator = new EndpointMetadataGenerator();
+    const endpointMetadata = endpointMetadataGenerator.generate({
+      agent: { version: kibanaPackageJson.version },
+    });
+    const now = overrides['@timestamp'] ?? new Date().toISOString();
+    const endpointAgentId = overrides?.agent?.id ?? this.seededUUIDv4();
+
+    return mergeAndReplaceArrays(
+      {
+        '@timestamp': now,
+        Endpoint: endpointMetadata.Endpoint,
+        agent: {
+          id: endpointAgentId,
+          type: 'endpoint',
+          version: kibanaPackageJson.version,
+        },
+        elastic: endpointMetadata.elastic,
+        host: endpointMetadata.host,
+        data_stream: {
+          dataset: 'endpoint.alerts',
+          namespace: 'default',
+          type: 'logs',
+        },
+        ecs: {
+          version: '1.4.0',
+        },
+        file: {
+          Ext: {
+            code_signature: [
+              {
+                subject_name: 'bad signer',
+                trusted: false,
+              },
+            ],
+            malware_classification: {
+              identifier: 'endpointpe',
```

Comment:
is this supposed to be `endpointype`?

Reply:
I'm not sure. I essentially "dumped" what a record looked like once the Endpoint rule ran and created alerts and just copied that. This is what I ended up with